beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -675,14 +682,18 @@ async fn availability_cache_maintenance_service<T: BeaconChainTypes>(
                     .fork_choice_read_lock()
                     .finalized_checkpoint()
                     .epoch;
+
+                // pre-fulu: use blob sidecar window, post-fulu: use data column window
+                let min_epochs = if chain.spec.is_peer_das_enabled_for_epoch(current_epoch) {
+                    chain.spec.min_epochs_for_data_column_sidecars_requests
+                } else {
+                    chain.spec.min_epochs_for_blob_sidecars_requests
+                };
```

Comment:
This will break in the transition. Consider a case where `min_epochs_for_blob_sidecars_requests = Inf` (never prune) and `min_epochs_for_data_column_sidecars_requests = 0` (prune instantly), after the Fulu fork you will prune all blobs which is incorrect.